src/methods/to.js

```diff
@@ -736,6 +736,8 @@ export class ToImplementation {
       let primValue = this.ToPrimitiveOrAbstract(realm, val, "string");
       if (primValue.getType() === StringValue) return primValue;
       str = this.ToStringPartial(realm, primValue);
+    } else if (val instanceof AbstractValue) {
+      return this.ToStringAbstract(realm, val);
     } else {
       throw realm.createErrorThrowCompletion(realm.intrinsics.TypeError, "unknown value type, can't coerce to string");
```

Comment:
Now that you also handle abstract values, this `realm.createErrorThrowCompletion` has potentially become dead code. Can you replace it with an `invariant(false);` and see if anything break? If not, then please include that change.

Reply:
We need to keep some sort of default case / sanity check, else `str` would be possibly uninitialized at the end of the function.

Reply:
But is that a feasible code path? If not, it should be guarded with `invariant(false);`. If yes, we are back to my original request in the issue: I don't think this should raise a JavaScript-level exception (where would the spec say so?), instead, it should produce a `CompilerDiagnostics` error and fail with `FatalError`.

Reply:
As far as I can tell the else is now an infeasible path and should be guarded by an invariant(false).

Reply:
Agreed, changing this.